### DISCUSSION TITLE
What happens if TCP message server connection failed?

### DISCUSSION THREAD

**Discussion by dbrosluc** [IonxSolutions/serilog-sinks-syslog#46] [General]
Hello,

Does Serilog provide any sort of error to indicate the TCP server connection failure?

**Comment by colin-anderson** [comment]
If there is a connection issue, the sink will automatically reconnect, and an error message will be logged via Serilog's [built-in diagnostic logging](https://github.com/serilog/serilog/wiki/Debugging-and-Diagnostics).

**Reply by dbrosluc** [reply]
Thank you Colin. I was hoping Serilog will return some sort of 401 error code or something to indicate there is error sending.

**Reply by colin-anderson** [reply]
Syslog isn't built on top of HTTP, so there is no concept of HTTP error codes or indeed anything similar - but we know if the TCP connection is broken or not.

**Reply by dbrosluc** [reply]
Hello Collin,

I am trying to write debug/diagnostic information by Enable SelfLog like this:
var file = File.CreateText("D:\\Serilog.error.log");
Serilog.Debugging.SelfLog.Enable(TextWriter.Synchronized(file));

But the log file isn't created. It works if I write it to debug  like this:
Serilog.Debugging.SelfLog.Enable(msg => System.Diagnostics.Debug.WriteLine(msg))

Do you have any idea what I might have not done to configure writing debug information to file?

**Reply by colin-anderson** [reply]
You can try this: https://stackoverflow.com/a/52772077
